src/solvers/dgsem_structured/subcell_limiters_2d.jl

```diff
@@ -56,8 +57,11 @@ function calc_bounds_twosided_interface!(var_min, var_max, variable, u, t, semi,
         for cell_y in axes(mesh, 2)
             element = linear_indices[begin, cell_y]
             for j in eachnode(dg)
-                u_outer = get_boundary_outer_state(boundary_conditions[1],
-                                                   cache, t, equations, dg,
+                Ja1 = get_contravariant_vector(1, contravariant_vectors, 1, j, element)
+                u_inner = get_node_vars(u, equations, dg, 1, j, element)
+                u_outer = get_boundary_outer_state(u_inner, t,
+                                                   boundary_conditions[1], Ja1, 1,
+                                                   mesh, equations, dg, cache,
```

Comment:
Please check if you need the outward-pointing normal vector here. If that's the case, you might need `-Ja1` for the first node... Or is that already taken into account with the fifth argument (1, 2)?

Reply:
That's a very good point. Thanks. I'll look into it and give some updates here, since it is not that clear to me right now how it is supposed to be.

Reply:
I start with some information about boundary conditions.
Slip wall already considers the sign change for StructuredMesh (see [here](https://github.com/trixi-framework/Trixi.jl/blob/9d8aac9c842d05865bef0bb7e50a6922710ed246/src/equations/compressible_euler_2d.jl#L339-L385)). 
Therefore, I implemented other boundary conditions simiarly (see [

Reply:
I start with some information about boundary conditions.
The sign change is already considered within the boundary condition for StructuredMesh (see [slip wall](https://github.com/trixi-framework/Trixi.jl/blob/9d8aac9c842d05865bef0bb7e50a6922710ed246/src/equations/compressible_euler_2d.jl#L339-L385) or [dirichlet](https://github.com/trixi-framework/Trixi.jl/blob/9d8aac9c842d05865bef0bb7e50a6922710ed246/src/equations/equations.jl#L166-L201)). 
Therefore, I implemented other boundary conditions similarly (see [double mach elixir in branch subcell limiting](https://github.com/bennibolm/Trixi.jl/blob/53881b7dd00af6ab028768397b974a64a695b534/examples/structured_2d_dgsem/elixir_euler_double_mach.jl#L51-L77).

Within `get_boundary_outer_state` it is considered as well. E.g.:

- slip_wall: Due to the current calculation, it seems to be indifferent to switch the sign of the normal vector (is that correct?)
- `boundary_condition_mixed_characteristic_wall`: slip wall part (see above); characteristics BC [it is considered in `characteristic_boundary_value_function`](https://github.com/bennibolm/Trixi.jl/blob/53881b7dd00af6ab028768397b974a64a695b534/src/equations/compressible_euler_2d.jl#L424]
- dirichlet & characteristics: passed to the sup routine and considered there.

So, it should be considered within the routines everywhere.

Reply:
I noticed that the boundary condition for TreeMesh (`orientation`, `direction`), StructuredMesh (`normal_direction`, `direction`) and P4estMesh (`normal_direction`) differ in the parameter. I probably can copy that for `get_boundary_condition outer_state` to make things easier to understand. So I will remove the parameter `direction` from the P4est routines. Also, I will not use the wrong routine unintentionally.

Reply:
I removed `direction` from the P4estMesh call in [baf2b30](https://github.com/trixi-framework/Trixi.jl/pull/1954/commits/baf2b307ff106ea3b55eb94f81c1c1abeebf13a5), as described above.

Reply:
Ok, good. That makes sense to me!

Reply:
> slip_wall: Due to the current calculation, it seems to be indifferent to switch the sign of the normal vector (is that correct?)

This is just a simple reflection across a plane that's perpendicular to the normal direction, right?... So it makes sense it doesn't matter if you provide the outward- or inward-pointing normal vector.